app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt

```diff
@@ -150,17 +151,26 @@ class SessionNotificationService : Service() {
             val intent = Intent(context, SessionNotificationService::class.java)
             intent.action = ACTION_START
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                context.startForegroundService(intent)
-            } else {
-                context.startService(intent)
-            }
+            // For #2901: The application is crashing due to the service not calling `startForeground`
```

Comment:
super-nits:
- "not calling `startForeground` before it times out"
- "speculative fix to decrease the time between starting the service": how? How does it work? I think it's important to say, "We're running after blocking UI thread methods like FocusApplication.onCreate (when clicking links) that could have delayed the `startForeground` invocation.`

Or similar, this wording isn't the best: late on a Friday. :)